gammapy/estimators/flux_estimator.py

```diff
@@ -0,0 +1,161 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import logging
+import numpy as np
+from astropy import units as u
+from gammapy.datasets import Datasets
+from gammapy.modeling.models import ScaleSpectralModel
+from gammapy.estimators import ParameterEstimator
+
+__all__ = ["FluxEstimator"]
+
+log = logging.getLogger(__name__)
+
+class FluxEstimator(ParameterEstimator):
+    """Flux estimator.
+
+    Estimates flux for a given list of datasets with their model in a given energy range.
+
+    To estimate the model flux the amplitude of the reference spectral model is
+    fitted within the energy range. The amplitude is re-normalized using the "norm" parameter,
+    which specifies the deviation of the flux from the reference model in this
+    energy range.
+
+    Parameters
+    ----------
+    datasets : list of `~gammapy.spectrum.SpectrumDataset`
+        Spectrum datasets.
+    source : str or int
+        For which source in the model to compute the flux points.
+    norm_min : float
+        Minimum value for the norm used for the fit statistic profile evaluation.
+    norm_max : float
+        Maximum value for the norm used for the fit statistic profile evaluation.
+    norm_n_values : int
+        Number of norm values used for the fit statistic profile.
+    norm_values : `numpy.ndarray`
+        Array of norm values to be used for the fit statistic profile.
+    sigma : int
+        Sigma to use for asymmetric error computation.
+    sigma_ul : int
+        Sigma to use for upper limit computation.
+    reoptimize : bool
+        Re-optimize other free model parameters.
+    """
+
+    def __init__(
+            self,
+            datasets,
+            source=0,
```

Comment:
Maybe we should make the `source` argument mandatory?

Reply:
Done